src/test.js

```diff
@@ -4,350 +4,329 @@ const autodux = require('./');
+      actual: counterState >= min && counterState <= max,
+      expected: true
     });
   }
 });
 
-describe('autodux().selectors', async assert => {
-  const { getValue } = createDux().selectors;
-
+describe('autodux({ … }).actions', async assert => {
   {
-    assert({
-      given: 'a property and value',
-      should: 'return a selector that knows its state slice',
-      actual: getValue({ counter: 3 }),
-      expected: 3
-    });
-  }
-
-  {
-    const initial = {
-      key1: 'value 1',
-      key2: 'value 2'
-    };
-    const store = {
-      slice: initial
-    };
-
     const {
-      selectors: { getKey1, getKey2 }
+      actions: { setUserName, setAge }
     } = autodux({
-      slice: 'slice',
-      initial
+      slice: 'user',
+      initial: {
+        userName: 'Anonymous',
+        age: 0
+      }
     });
 
-    const actual = {
-      key1: getKey1(store),
-      key2: getKey2(store)
-    };
+    const userName = 'Freddie';
+    const age = 23;
 
     assert({
-      given: 'a property and value',
-      should: 'expose a selector for each key in the initial state',
-      actual,
-      expected: initial
+      given: "'autodux' is called without 'actions'",
+      should:
+        'contain correct action creators for each key in the initial state',
+      actual: [setUserName(userName), setAge(age)],
+      expected: [
+        {
+          type: 'user/setUserName',
+          payload: userName
+        },
+        {
+          type: 'user/setAge',
+          payload: age
+        }
+      ]
     });
   }
+});
 
+describe('autodux({ actions: … }).reducer', async assert => {
   {
-    const initial = {
-      userName: 'Anonymous',
-      avatar: 'anonymous.png'
-    };
-    const store = {
-      user: initial
-    };
-
     const {
-      selectors: { getUser }
-    } = autodux({
-      slice: 'user',
-      initial: {
-        userName: 'Anonymous',
-        avatar: 'anon.png'
-      }
-    });
+      actions: { increment, decrement },
+      reducer,
+      initial
+    } = createCounterDux();
 
     assert({
-      given: 'selectors',
-      should: 'expose a selector for the entire reducer state',
-      actual: getUser(store),
-      expected: initial
+      given: "'autodux' is called with 'actions'",
+      should: 'return reducer that switches correctly',
+      actual: [increment(), increment(), increment(), decrement()].reduce(
+        reducer,
+        initial
+      ),
+      expected: 2
     });
   }
 
   {
-    const { getStore } = createDux().selectors;
+    const {
+      actions: { increment, multiply },
+      reducer,
+      initial
+    } = createCounterDux();
 
     assert({
-      given: 'a store',
-      should: 'pass entire store as a second parameter to selectors',
-      actual: getStore({ counter: 3, foo: 'bar' }),
-      expected: { counter: 3, foo: 'bar' }
+      given: "'autodux' is called with 'actions'",
+      should: 'return reducer that considers delivering action payloads',
```

Comment:
I don't understand this `should` description. What does "considers delivering action payloads" mean?

Reply:
It should've been "_return reducer that considers action payload_". I was trying to check that `autodux` takes into account (_considers_) that action payload is passed to reducer function as the second parameter.

Probably the following is going to read better: "_return reducer that receives action payload as the second parameter_".

Reply:
Updated this description to "_return reducer that receives action payload as the second parameter_".